### PR TITLE
DevinDavisExamAnswers

### PR DESCRIPTION
**Pull request by DevinDavis**

### CODE REVIEW

File: Loops/Program.cs [hunk 1]
```diff
@@ -58,10 +58,16 @@ class Program
             new Student {FirstName = "Soroush (Looloosh)", LastName="Farazmand" },
             new Student {FirstName = "Monkey", LastName="Black" }
         };
-
+        //1. Write code to print all the names of all the studnets in the collection above.
         static void Main(string[] args)
         {
-            //1. Write code to print all the names of all the studnets in the collection above.
```
Comment (BillWagner) [inline Loops/Program.cs:64]: Good work.

**Comment by BillWagner** [comment]
Here's my comments on the written portion:

Model: Good start. I think you need to understand primary key and foreign key better.

SQL Questions:
1.Correct.
2,3: Close. You've got the columns right, and the 2nd question is correct. The difference between an Inner Join and a Left Join is that the inner join only returns those customer rows that have orders. The left join returns all customers, even those with no orders.

EF Questions:
Materialization: Correct.
Change Tracking: Correct
Object Identities: Correct.
Code First Model: That's a model, but it was copied from the Blog project. I was hoping to see you create your own model for Project and Task.

OOP Questions:
concepts: Correct
Behaviors: Correct
Overriding: Correct
Inheritance: Very good.

Abstract class: Good.

MVC Questions:

URL: You were correct. I was looking for the URL that would map to a controller. This would be something like:
http://mysite.com/Catalog/Specifications/42 (or any other number). 
But, without the ID it would still route to the same controller and method.

HTTP VERB: This would be GET. Remember how we made controllers that have two methods that handle the same URL: one would display the view, and one would process user input. The one that displays the view processes an HTTP GET. The one that handles user input processes an HTTP PUT.

View questions:

Model: The model is an IEnumerable&lt;MVCGuidedLab.Color&gt;
It's a sequence (because it's an enumerable)
Properties: Correct. You could also include Id, because the Id is used in the ActionLink directives at the bottom.

**Comment by BillWagner** [comment]
Some of the concepts are starting to come together. 

I am concerned about how long the written section took, and how little code you wrote. I really want to see you spend as much time the next four weeks on coding exercises as possible. 

I encourage you to try the remaining coding problems, and see how well you can do without the time pressure.  The extra practice would also help you get more confident in C# coding exercises in general.

**Comment by DevinDavis** [comment]
Thanks  Bill,

We went over all of the exam questions today and I was able to see where I
mad mistakes. I think I have a better understanding now as your comments
where also most helpful and straight forward. I  am working on the coding
problems this evening and I'm pushing myself to correctly complete them
all. I am determine to really get c# coding as the concepts are now
starting to make sense to me.

Thanks again,

Devin Davis
dndavis1111@gmail.com

**Comment by BillWagner** [comment]
Devin,

I'm sure you will keep improving. Just keep at it, and we'll keep feeding content your way.

**Pull request closed without merging**